docs/sources/flow/reference/components/discovery.marathon.md

```diff
@@ -0,0 +1,141 @@
+---
+canonical: https://grafana.com/docs/agent/latest/flow/reference/components/discovery.marathon/
+title: discovery.marathon
+---
+
+# discovery.marathon
+
+`discovery.marathon` allows you to retrieve scrape targets from [Marathon's](https://mesosphere.github.io/marathon/) Service API.
+
+## Usage
+
+```river
+discovery.marathon "LABEL" {
+  servers = [MARATHON_SERVER1, MARATHON_SERVER2...]
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+| Name               | Type           | Description                                                  | Default                 | Required |
+| ------------------ | -------------- | ------------------------------------------------------------ | ----------------------- | -------- |
+| `servers`          | `list(string)` | List of Marathon servers.                                    | `http://localhost:4646` | no       |
+| `refresh_interval` | `duration`     | Interval at which to refresh the list of targets.            | `"30s"`                 | no       |
+| `auth_token`       | `secret`       | Auth token to authenticate with.                             |                         | no       |
+| `auth_token_file`  | `string`       | File containing an auth token to authenticate with.          |                         | no       |
+| `proxy_url`        | `string`       | HTTP proxy to proxy requests through.                        |                         | no       |
+| `follow_redirects` | `bool`         | Whether redirects returned by the server should be followed. | `true`                  | no       |
+| `enable_http2`     | `bool`         | Whether HTTP2 is supported for requests.                     | `true`                  | no       |
```

Comment:
Off-topic, but I noticed some proxy-related options missing from the upstream configuration. 

Nothing to do in this PR, but I opened an [issue](https://github.com/grafana/agent/issues/5110) to keep track of it.